### PR TITLE
logout and hasPermission updates

### LINKED ISSUE #1122: logout and login inline page behavior
For bugs, please fill out the three following sections completely

### Expected behavior

Whenever a user logs out, we should redirect to the home page.
A login box should display and message should display when a user access a url they are not authorized to access.
When a login box is displayed inline, a user should be able to login on first try.
When a user logins the correct page should display, not the login.

### Actual Behavior

The user is shown a `Oops! You don't have permission to view this content.` and a login box.

The login does not appear to work on the first try, but subsequent attempts do.

Once you login the page displayed is still the login, although admin menu bars etc display.

### Steps to Reproduce the Behavior

A) Login in as admin view dashboard, log out.
B)  go to: 
`http://localhost:3000/reaction/dashboard`
and login.

### PR DESCRIPTION
**Pull request by aaronjudd**
- resolves #1122
- refactors client hasPermissions to wait for a userId if one isn’t
immediately found
- intentionally not redirecting to home page (not sure if that’s the
best behavior, better to have login?)
- adds subscription manager to a few more collections

To test: login and logout.. all over the place.

### CODE REVIEW

**Comment by zenweasel** [comment]
I'm seeing this error on first load after reset:

```
Exception in onLogin callback: TypeError: Cannot read property 'length' of undefined
    at parse (http://localhost:3000/packages/kadira_flow-router-ssr.js?hash=cb5097fd0bf72f2e3de9237b1dacbc3e6d545e15:1976:18)
    at Function.compile (http://localhost:3000/packages/kadira_flow-router-ssr.js?hash=cb5097fd0bf72f2e3de9237b1dacbc3e6d545e15:1995:27)
    at Router.path (http://localhost:3000/packages/kadira_flow-router-ssr.js?hash=cb5097fd0bf72f2e3de9237b1dacbc3e6d545e15:114:33)
    at Router.go (http://localhost:3000/packages/kadira_flow-router-ssr.js?hash=cb5097fd0bf72f2e3de9237b1dacbc3e6d545e15:738:23)
    at http://localhost:3000/packages/kadira_flow-router-ssr.js?hash=cb5097fd0bf72f2e3de9237b1dacbc3e6d545e15:604:16
    at withValue (http://localhost:3000/packages/meteor.js?hash=ae8b8affa9680bf9720bd8f7fa112f13a62f71c3:1077:17)
    at Router.reload (http://localhost:3000/packages/kadira_flow-router-ssr.js?hash=cb5097fd0bf72f2e3de9237b1dacbc3e6d545e15:603:23)
    at http://localhost:3000/app/app.js?hash=f870536833dbb275bf7364b36520e5aeabf226b6:34590:14
    at runAndHandleExceptions (http://localhost:3000/packages/callback-hook.js?hash=fff7fdef0707c85900b21f766a4b6c65bf278ff4:162:24)
    at http://localhost:3000/packages/callback-hook.js?hash=fff7fdef0707c85900b21f766a4b6c65bf278ff4:169:12
```

All subsequent requests are fine. It doesn't appear to be happening in `development`

File: client/modules/core/main.js [hunk 1]
```diff
@@ -77,56 +78,95 @@ export default {
...
-      for (let key in sellerShopPermissions) {
-        if (key) {
-          let shop = sellerShopPermissions[key];
-          if (Roles.userIsInRole(userId, permissions, shop)) {
-            return true;
-          }
-        }
+    if (Meteor.loggingIn() === false) {
+      //
+      // this userId check is happens because when logout
```
Comment (zenweasel) [inline client/modules/core/main.js]: typo in comment?

Comment (aaronjudd) [inline client/modules/core/main.js]: You want comments that make sense to everyone?

**Comment by aaronjudd** [comment]
@zenweasel ok that error should be resolved.

**Comment by zenweasel** [comment]
👍

**Comment by jshimko** [comment]
👍 

It all works for me.  However, I do feel like not having a logout redirect is somewhat unexpected behavior.  I think I'm just used to redirects going to a public facing unauthenticated page (home page or whatever).  Reloading the same route and showing a permissions error makes it feel like something has gone wrong to me.

Either way, main problem fixed!

**Comment by aaronjudd** [comment]
@jshimko yeah I was a little undecided on that..  the redirect is easy of course, but my original implementation was meant to show the login, so I was happy to get the original intent working as expected.  There is a couple interesting cases, for instance when viewing order details and you logout you just get a blank screen.  On the other hand.. if we had an automatic logout (something in the future?).. I'd want to be logging back in where I logged out...  

Feels like there's a couple remaining "handling" issues here that we need to discuss the workflow, but this is much better... so I'm merging.

**Pull request merged**